sdcm/sct_runner.py

```diff
@@ -1279,7 +1271,18 @@ def _manage_runner_keep_tag_value(utc_now: datetime,
             sct_runner_info.keep = new_keep_value
         return sct_runner_info
 
-    LOGGER.info("No changes to make to runner tags.")
+    if sct_runner_info.logs_collected:
+        if not dry_run:
+            sct_runner_info.sct_runner_class.set_tags(sct_runner_info, {"keep": "0", "keep-action": "terminate"})
```

Comment:
this code is running on this runner ? isn't there a possibility it would get cleared right away ?
maybe 1 is a safer option ?

why in perf tests are not getting into the first if ?

Reply:
that's the intention - clear right away (it's the last step in the pipeline, part of `clean_sct_runners` method) if logs are collected.

Perf tests often fail with an error event, and not getting into the first if. The second seemed to have broken logic: `if not timeout_flag` - meaning only if not timed out (basically always - our tests rarely timeout recently) we prolong by another 6 hours which makes no sense.

Two changes:
1. invert logic behind timeout_flag - when test timed out, add another 6h just in case some more info are required to be obtained from runner (e.g. see which process frozen).
2. Logic was changed to simpler one: if logs are collected, terminate runner, otherwise prolong duration by 48h.